### DISCUSSION TITLE
bug: modern fortran extension cannot find fprettify even when pointed directly to it (Linux)

### DISCUSSION THREAD

**Discussion by sjsmith757** [krvajal/vscode-fortran-support#1144] [Q&A - deprecated see Fortran Discourse]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### I have tried the Pre-Release of the extension.

- [X] I have replicated the Issue in the latest Pre-Release

### Description

Modern Fortran v3.4.2024090901
fortls v3.1.2
fprettify v0.3.7

Steps to reproduce:
1) Install fortls and fprettify via conda
2) Install Modern Fortran pre-release
3) load the conda environment
3b) Optional, modify settings.json to point to the fprettify binary
4) try to format a file

Currently, I am unable to format files using fprettify through the extension. I can format files without issue using `fprettify foo.f90`. I have tried using both the default location which is in my `$PATH`, or modifying the settings.json `fortran.formatting.path`, but it cannot find fprettify in either case, even though it is there and working fine. According to the Log Channel, it is even seeing my modifications to the settings.json (I have tried restarting VS Code after modifying the file). However, the attempt to install (even though it is already there) always fails (because it is installed). And so the problem is never resolved, and the formatter never runs.

### Screenshots

![image](https://github.com/user-attachments/assets/74add512-ba76-473b-be96-329558de0950)

### Expected Behaviour

File to be formatted using fprettify

### Version of Modern Fortran

v3.4.2024090901

### Version of Visual Studio Code

v1.93.0

### Platform and Architecture

Linux

**Comment by gnikit** [comment]
Conda and generally all virtualisations of Python are not supported by Modern Fortran since we would have to include similar logic to the ms-python extension, and I don't have the bandwidth to do this at the moment, although it is part of my TODOs for the extension. Have you tried using a globally discoverable (i.e. in your `$PATH`) installation of the respective tools?
The workaround of working with whatever Python virtualisation is to launch vscode from a terminal that has the environment activated, e.g.

```sh
conda activate lf
code .
```

This would make fprettify, fortls, etc. discoverable.
